### PR TITLE
Adding a Design doc template

### PR DESCRIPTION
**Pull request by sanathkr**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,5 +2,12 @@
 
 *Description of changes:*
 
+*Checklist:*
+
+- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
+- [ ] Write unit tests
+- [ ] Write/update integration tests
```
Comment (jfuss) [inline .github/PULL_REQUEST_TEMPLATE.md]: What about functional?

File: DEVELOPMENT_GUIDE.rst [hunk 2]
```diff
@@ -133,9 +133,25 @@ best practices that we have learnt over time.
...
 
+Design Document
+---------------
+
+A design document is a written description of the feature/capability you are building. We have a
+`design document template`_ to help you quickly fill in the blanks and get you working quickly. We encourage you to
+write a design document for any feature you write, but for some types of features we definitely require a design
+document to proceed with implementation.
+
+**When do you need a design document?**
```
Comment (jfuss) [inline DEVELOPMENT_GUIDE.rst]: This is super clear. Really like how you defined this!

File: designs/_template.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+Documentation Changes
+---------------------
+
+Open Issues
+-----------
+
+Task Breakdown
+--------------
+- [x] Send a Pull Request with this design document (PR #123)
```
Comment (jfuss) [inline designs/_template.rst]: Are the PR # and Issue # relevant to these items? #123 isn't a design doc PR.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (sanathkr) [reply .github/PULL_REQUEST_TEMPLATE.md]: added

File: designs/_template.rst (hunk 3 above)
Comment (sanathkr) [reply designs/_template.rst]: removed

**Pull request merged**